src/HermesWorker/index.js

```diff
@@ -228,12 +229,14 @@ class HermesWorker {
     /**
      * Is called from worker in case of thrown error
      *
-     * TODO: Improve error handling
-     *
      * @param {any} error
      */
     _onWorkerError(error) {
-        console.error(error);
+        if (!this.isLoaded) {
+            this._loadError = error;
+            this._loadedPromises.forEach(({ reject }) => reject(error));
+        }
+        Object.values(this._pendingsCalls).forEach(defer => defer.reject(error));
```

Comment:
Hello @jbghoul,

I come after the battle! 
Here, we stop all the executions in progress, which is not necessary in case there are several instances of the same code.

Let me explain, if you run HermesWorker with "threadInstances" set to 2, there will be two workers to run! If worker 1 breaks, worker two still works, so you don't need to terminate the executions in the second worker.

I think we can send the worker (which has errors) in _onWorkerError, and store the worker (or an identifier) in _pendingsCalls, so we can stop the executions only in the first worker